src/tls/workload.rs

```diff
@@ -163,19 +171,12 @@ pub struct OutboundConnector {
 }
 
 impl OutboundConnector {
-    pub async fn connect(
-        self,
-        stream: TcpStream,
-    ) -> Result<client::TlsStream<TcpStream>, io::Error> {
-        let dest = ServerName::IpAddress(
-            stream
-                .peer_addr()
-                .expect("peer_addr must be set")
-                .ip()
-                .into(),
-        );
+    pub async fn connect<IO>(self, stream: IO) -> Result<client::TlsStream<IO>, io::Error>
+    where
+        IO: AsyncRead + AsyncWrite + Unpin,
+    {
         let c = tokio_rustls::TlsConnector::from(self.client_config);
-        c.connect(dest, stream).await
+        c.connect(DUMMY_DOMAIN.clone(), stream).await
```

Comment:
Can you please revert this. We should NOT be sending an SNI.

rustls makes this kind of annoying, but if you just set an IP it will not send anything. We can put 0.0.0.0 if you want.

Putting a dummy IP != putting a dummy domain